android/src/main/java/com/swmansion/reanimated/layoutReanimation/Snapshot.java

```diff
@@ -115,6 +126,28 @@ public Snapshot(View view) {
     }
     originXByParent = view.getLeft();
     originYByParent = view.getTop();
+    borderRadius = getBorderRadius(view);
+  }
+
+  private float getBorderRadius(View view) {
+    if (view.getBackground() != null) {
+      Drawable background = view.getBackground();
+      if (background instanceof ReactViewBackgroundDrawable) {
+        return ((ReactViewBackgroundDrawable) background).getFullBorderRadius();
+      }
+    } else if (view instanceof ReactImageView) {
```

Comment:
I wonder if instead of relying on RN internals here and above we could just use background drawable and extract corner radii using `getOutline().getRadius()`

Not a huge fan of adding this type of dependencies, especially that it only gives us support for built-in image and view components. If we need to change visibility of fields this should be done only once per class anyways, so it'd be good to try and avoid more reflection calls than necessary

Reply:
I have tried to use `getOutline().getRadius()` but it doesn’t work. The `getRadius()` always returns `-Infinity`. 

According to Android documentation:(https://developer.android.com/reference/android/graphics/Outline#getRadius())
>Returns the rounded rect radius, if set, **or a value less than 0** if a path has been set via `setPath(android.graphics.Path)`

Unfortunately React Native set border radius via `setPath()` method.  [ReactViewBackgroundDrawable .drawRoundedBackgroundWithBorders() source code](https://github.com/facebook/react-native/blob/v0.72.0-rc.6/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/view/ReactViewBackgroundDrawable.java#L341-L346)
```java
private void drawRoundedBackgroundWithBorders(Canvas canvas) {
  updatePath(); // <-- calls setPath()
  canvas.save();
  // Clip outer border
  canvas.clipPath(mOuterClipPathForBorderRadius, Region.Op.INTERSECT);
```

Every ReactNative component uses `ReactViewBackgroundDrawable` to set border radius, except ReactImageView.

[ReactViewGroup setBorderRadius() source code](https://github.com/facebook/react-native/blob/v0.72.0-rc.6/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/view/ReactViewGroup.java#L319-L322)
```java
public void setBorderRadius(float borderRadius) {
  ReactViewBackgroundDrawable backgroundDrawable = getOrCreateReactViewBackground();
  backgroundDrawable.setRadius(borderRadius);
}
```

[ReactImageView setBorderRadius() source code](https://github.com/facebook/react-native/blob/v0.72.0-rc.6/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/image/ReactImageView.java#L261-L266)
```java
public void setBorderRadius(float borderRadius) {
  if (!FloatUtil.floatsEqual(mBorderRadius, borderRadius)) {
    mBorderRadius = borderRadius;
    mIsDirty = true;
  }
}
```

I have tried many different approaches but this one with `ReactViewBackgroundDrawable` works best for me.

---

I have reduced the number of reflection calls in 2cc22c61e68bf99a690cbff239d216a893b4f5f3